src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorStopRepresentingClient.java

```diff
@@ -0,0 +1,214 @@
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} about to submit callback invoked for Case Id: {}", SOLICITOR_STOP_REPRESENTING_CLIENT, details.getId());
+
+        final boolean isRepresentingApplicant1 = isApplicant1Solicitor(details.getId());
+        final UserRole orgPolicyRole = isRepresentingApplicant1 ? APPLICANT_1_SOLICITOR : APPLICANT_2_SOLICITOR;
+        final List<String> rolesToRemove = isRepresentingApplicant1
+            ? List.of(CREATOR.getRole(), APPLICANT_1_SOLICITOR.getRole())
+            : List.of(APPLICANT_2.getRole(), APPLICANT_2_SOLICITOR.getRole());
+        final Function<CaseData, Applicant> applicant = isRepresentingApplicant1 ? CaseData::getApplicant1 : CaseData::getApplicant2;
+
+        recordNoticeOfChangePartyInCaseData(details, isRepresentingApplicant1);
+
+        removeSolicitorDetailsFromCaseData(applicant.apply(details.getData()), orgPolicyRole);
+
+        changeOfRepresentativeService.buildChangeOfRepresentative(
+            details.getData(),
+            beforeDetails.getData(),
+            ChangeOfRepresentationAuthor.SOLICITOR_STOP_REPRESENTING_CLIENT.getValue(),
+            isRepresentingApplicant1
+        );
+
+        NoticeType.ORG_REMOVED.applyNoticeOfChange(
+            applicant.apply(details.getData()),
+            applicant.apply(beforeDetails.getData()),
+            rolesToRemove,
+            orgPolicyRole.getRole(),
+            details,
+            noticeOfChangeService
+        );
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(details.getData())
+            .build();
+    }
+
+    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
+                                               CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} submitted callback invoked for Case Id: {}", SOLICITOR_STOP_REPRESENTING_CLIENT, details.getId());
+
+        final CaseData data = details.getData();
+        boolean wasRepresentingApplicant1 = data.getNoticeOfChange().getWhichApplicant() == WhichApplicant.APPLICANT_1;
+
+        notificationDispatcher.sendNOC(nocSolRemovedSelfNotifications, details.getData(),
+            beforeDetails.getData(), details.getId(), wasRepresentingApplicant1, NoticeType.ORG_REMOVED);
+
+        if (data.getApplicationType() == ApplicationType.SOLE_APPLICATION) {
+            final var applicant = wasRepresentingApplicant1 ? data.getApplicant1() : data.getApplicant2();
+            generateCaseInvite(data, wasRepresentingApplicant1, applicant);
+            notificationDispatcher.sendNOCCaseInvite(nocSolRemovedSelfNotifications, details.getData(), details.getId(),
+                wasRepresentingApplicant1);
+        }
```

Comment:
If I'm not mistaken, across the NoC events (caseworkerNoticeOfChange, solicitorStopRepresentingClient), we send the case invite whenever the notice type is `ORG_REMOVED` and the application type is `sole`. As most of the logic for NoC notifications is in `sendNOC`, I'm thinking about whether it would be beneficial to remove `sendNOCCaseInvite` and move the logic for the case invite notification into the `sendNOC` method, allowing us consolidate the notification logic into 1 method and remove duplication from the NoC events?

```suggestion
        notificationDispatcher.sendNOC(nocSolRemovedSelfNotifications, details.getData(),
            beforeDetails.getData(), details.getId(), wasRepresentingApplicant1, NoticeType.ORG_REMOVED);
```